src/utils/formatNumbers.ts

```diff
@@ -1,244 +1,302 @@
-  { upperBound: 10000, formatter: SIX_SIG_FIGS_TWO_DECIMALS },
-  { upperBound: Infinity, formatter: TWO_DECIMALS },
+  { exact: 0, hardCodedInput: { input: 0 }, formatterOptions: NO_DECIMALS },
+  {
+    upperBound: 0.00001,
+    hardCodedInput: { input: 0.00001, prefix: '<' },
+    formatterOptions: FIVE_DECIMALS_MAX_TWO_DECIMALS_MIN,
+  },
+  { upperBound: 1, formatterOptions: FIVE_DECIMALS_MAX_TWO_DECIMALS_MIN },
+  { upperBound: 10000, formatterOptions: SIX_SIG_FIGS_TWO_DECIMALS },
+  { upperBound: Infinity, formatterOptions: TWO_DECIMALS },
 ]
 
 const swapTradeAmountFormatter: FormatterRule[] = [
-  { exact: 0, formatter: '0' },
-  { upperBound: 0.1, formatter: SIX_SIG_FIGS_NO_COMMAS },
-  { upperBound: 1, formatter: FIVE_DECIMALS_MAX_TWO_DECIMALS_MIN_NO_COMMAS },
-  { upperBound: Infinity, formatter: SIX_SIG_FIGS_TWO_DECIMALS_NO_COMMAS },
+  { exact: 0, hardCodedInput: { input: 0 }, formatterOptions: NO_DECIMALS },
+  { upperBound: 0.1, formatterOptions: SIX_SIG_FIGS_NO_COMMAS },
+  { upperBound: 1, formatterOptions: FIVE_DECIMALS_MAX_TWO_DECIMALS_MIN_NO_COMMAS },
+  { upperBound: Infinity, formatterOptions: SIX_SIG_FIGS_TWO_DECIMALS_NO_COMMAS },
 ]
 
 const swapPriceFormatter: FormatterRule[] = [
-  { exact: 0, formatter: '0' },
-  { upperBound: 0.00001, formatter: '<0.00001' },
+  { exact: 0, hardCodedInput: { input: 0 }, formatterOptions: NO_DECIMALS },
+  {
+    upperBound: 0.00001,
+    hardCodedInput: { input: 0.00001, prefix: '<' },
+    formatterOptions: FIVE_DECIMALS_MAX_TWO_DECIMALS_MIN,
+  },
   ...swapTradeAmountFormatter,
 ]
 
 const fiatTokenDetailsFormatter: FormatterRule[] = [
-  { exact: 0, formatter: '$0.00' },
-  { upperBound: 0.00000001, formatter: '<$0.00000001' },
-  { upperBound: 0.1, formatter: THREE_SIG_FIGS_USD },
-  { upperBound: 1.05, formatter: THREE_DECIMALS_USD },
-  { upperBound: 1e6, formatter: TWO_DECIMALS_USD },
-  { upperBound: Infinity, formatter: SHORTHAND_USD_TWO_DECIMALS },
+  { exact: 0, hardCodedInput: { input: 0 }, formatterOptions: TWO_DECIMALS_CURRENCY },
+  {
+    upperBound: 0.00000001,
+    hardCodedInput: { input: 0.00000001, prefix: '<' },
+    formatterOptions: ONE_SIG_FIG_CURRENCY,
+  },
+  { upperBound: 0.1, formatterOptions: THREE_SIG_FIGS_CURRENCY },
+  { upperBound: 1.05, formatterOptions: THREE_DECIMALS_CURRENCY },
+  { upperBound: 1e6, formatterOptions: TWO_DECIMALS_CURRENCY },
+  { upperBound: Infinity, formatterOptions: SHORTHAND_CURRENCY_TWO_DECIMALS },
 ]
 
 const fiatTokenPricesFormatter: FormatterRule[] = [
-  { exact: 0, formatter: '$0.00' },
-  { upperBound: 0.00000001, formatter: '<$0.00000001' },
-  { upperBound: 1, formatter: THREE_SIG_FIGS_USD },
-  { upperBound: 1e6, formatter: TWO_DECIMALS_USD },
-  { upperBound: 1e16, formatter: SHORTHAND_USD_TWO_DECIMALS },
-  { upperBound: Infinity, formatter: SEVEN_SIG_FIGS__SCI_NOTATION_USD },
+  { exact: 0, hardCodedInput: { input: 0 }, formatterOptions: TWO_DECIMALS_CURRENCY },
```

Comment:
for the formatters on `exact: 0`, do you need the hardcodedInput of `0` as well?

Reply:
lol good point no it doesn't